src/main/java/com/troblecodings/signals/items/Placementtool.java

```diff
@@ -83,19 +83,29 @@ public InteractionResult useOn(final UseOnContext context) {
         final Signal signal = getObjFromID(wrapper.getInteger(BLOCK_TYPE_ID));
         final List<SEProperty> properties = signal.getProperties();
         final Map<SEProperty, String> signalProperties = new HashMap<>();
-        properties.forEach(property -> {
+        int cost = signal.getDefaultDamage();
+
+        for (final SEProperty property : properties) {
             final String name = property.getName();
-            if (wrapper.contains(name) && !property.isChangabelAtStage(ChangeableStage.APISTAGE)) {
-                signalProperties.put(property, wrapper.getString(name));
-            }
-            if (property.isChangabelAtStage(ChangeableStage.APISTAGE) && wrapper.contains(name)) {
-                signalProperties.put(property, property.getDefault());
-            }
-            if (property.isChangabelAtStage(ChangeableStage.APISTAGE_NONE_CONFIG)
+            if (wrapper.contains(name)) {
+                cost += property.getItemDamage();
+
+                if (!property.isChangabelAtStage(ChangeableStage.APISTAGE)) {
+                    signalProperties.put(property, wrapper.getString(name));
+                } else if (property.isChangabelAtStage(ChangeableStage.APISTAGE)) {
+                    signalProperties.put(property, property.getDefault());
+                }
+            } else if (property.isChangabelAtStage(ChangeableStage.APISTAGE_NONE_CONFIG)
                     && property.testMap(signalProperties)) {
                 signalProperties.put(property, property.getDefault());
+            } else if (property.isChangabelAtStage(ChangeableStage.GUISTAGE)) {
+                signalProperties.put(property, property.getDefault());
             }
-        });
+        }
+
+        final ItemStack item = context.getItemInHand();
+        item.hurtAndBreak(cost, player, (user) -> user.broadcastBreakEvent(context.getHand()));
```

Comment:
Check that cost is not negative!
```suggestion
        item.hurtAndBreak(Math.abs(cost), player, (user) -> user.broadcastBreakEvent(context.getHand()));
```